<!--
  1 failure:  :disappointed_rel...
  2 warning:  There is no linke..., :mag: Can't find ...
  3 messages
  
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>There is no linked JIRA ticket in the PR body. Please include the URL of the relevant JIRA ticket. If you need to, you may file a ticket on <a href="https://jira.appcelerator.org/secure/CreateIssue!default.jspa">JIRA</a></td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  :mag: Can't find junit reports at `./junit.*.xml`, skipping generating JUnit Report.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:tada: Another contribution from our awesome community member, drauggres! Thanks again for helping us make Titanium SDK better. :thumbsup:</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  ```
> titanium-mobile@9.0.0 test /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-11352
> npm run ios-sanity-check && npm run lint


> titanium-mobile@9.0.0 ios-sanity-check /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-11352
> ./build/scons check-ios-toplevel


> titanium-mobile@9.0.0 lint /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-11352
> npm-run-all --parallel lint:**


> titanium-mobile@9.0.0 lint:android /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-11352
> clang-format-lint $npm_package_config_format_android


> titanium-mobile@9.0.0 lint:docs /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-11352
> tdoc-validate ./apidoc


> titanium-mobile@9.0.0 lint:ios /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-11352
> clang-format-lint $npm_package_config_format_ios


> titanium-mobile@9.0.0 lint:js /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-11352
> eslint .

[INFO] Global: OK!
[INFO] Global.JSON: OK!
[INFO] Global.String: OK!
[INFO] Global.console: OK!
[INFO] Modules.Cloud.ACLs: OK!
[INFO] CloudACLsResponse: OK!
[INFO] CloudACLsCheckResponse: OK!
[INFO] Modules.Cloud.Chats: OK!
[INFO] CloudChatsResponse: OK!
[INFO] CloudChatGroupsResponse: OK!
[INFO] Modules.Cloud.Checkins: OK!
[INFO] CloudCheckinsResponse: OK!
[INFO] Modules.Cloud.Clients: OK!
[INFO] CloudClientsResponse: OK!
[INFO] Modules.Cloud: OK!
[INFO] CloudStreamProgress: OK!
[INFO] CloudResponse: OK!
[INFO] Modules.Cloud.Emails: OK!
[INFO] CloudEmailsResponse: OK!
[INFO] Modules.Cloud.Events: OK!
[INFO] CloudEventsResponse: OK!
[INFO] CloudEventOccurrencesResponse: OK!
[INFO] Modules.Cloud.Files: OK!
[INFO] CloudFilesResponse: OK!
[INFO] Modules.Cloud.Friends: OK!
[INFO] CloudFriendsResponse: OK!
[INFO] CloudFriendRequestsResponse: OK!
[INFO] Modules.Cloud.GeoFences: OK!
[INFO] CloudGeoFenceResponse: OK!
[INFO] Modules.Cloud.KeyValues: OK!
[INFO] CloudKeyValuesResponse: OK!
[INFO] Modules.Cloud.Likes: OK!
[INFO] CloudLikesResponse: OK!
[INFO] Modules.Cloud.Messages: OK!
[INFO] CloudMessagesResponse: OK!
[INFO] Modules.Cloud.Objects: OK!
[INFO] CloudObjectsResponse: OK!
[INFO] Modules.Cloud.PhotoCollections: OK!
[INFO] CloudPhotoCollectionsResponse: OK!
[INFO] CloudPhotoCollectionsPhotosResponse: OK!
[INFO] Modules.Cloud.Photos: OK!
[INFO] CloudPhotosResponse: OK!
[INFO] Modules.Cloud.Places: OK!
[INFO] CloudPlacesResponse: OK!
[INFO] Modules.Cloud.Posts: OK!
[INFO] CloudPostsResponse: OK!
[INFO] Modules.Cloud.PushNotifications: OK!
[INFO] CloudPushNotificationsResponse: OK!
[INFO] CloudPushNotificationsQueryChannelResponse: OK!
[INFO] CloudPushNotificationsShowChannelResponse: OK!
[INFO] CloudPushNotificationsQueryResponse: OK!
[INFO] Modules.Cloud.PushSchedules: OK!
[INFO] CloudPushSchedulesResponse: OK!
[INFO] Modules.Cloud.Reviews: OK!
[INFO] CloudReviewsResponse: OK!
[INFO] Modules.Cloud.SocialIntegrations: OK!
[INFO] CloudSocialIntegrationsResponse: OK!
[INFO] Modules.Cloud.Statuses: OK!
[INFO] CloudStatusesResponse: OK!
[INFO] Modules.Cloud.Users: OK!
[INFO] CloudUsersSecureDialog: OK!
[INFO] CloudUsersResponse: OK!
[INFO] CloudUsersSecureResponse: OK!
[INFO] Modules.CloudPush: OK!
[INFO] CloudPushNotificationConfig: OK!
[INFO] CloudPushNotificationSuccessArg: OK!
[INFO] CloudPushNotificationErrorArg: OK!
[INFO] Modules: OK!
[INFO] Titanium.API: OK!
[INFO] Titanium.Accelerometer: OK!
[INFO] Titanium.Analytics: OK!
[INFO] Titanium.Android.ActionBar: OK!
[INFO] Titanium.Android.Activity: OK!
[INFO] ActivityResult: OK!
[INFO] Titanium.Android: OK!
[INFO] ServiceIntentOptions: OK!
[INFO] RequestPermissionAccessResult: OK!
[INFO] Titanium.Android.BigPictureStyle: OK!
[INFO] Titanium.Android.BigTextStyle: OK!
[INFO] Titanium.Android.BroadcastReceiver: OK!
[INFO] Titanium.Android.Calendar.Alert: OK!
[INFO] Titanium.Android.Calendar: OK!
[INFO] CalendarPermissionResponse: OK!
[INFO] Titanium.Android.Calendar.Calendar: OK!
[INFO] Titanium.Android.Calendar.Event: OK!
[INFO] Titanium.Android.Calendar.Reminder: OK!
[INFO] Titanium.Android.Intent: OK!
[INFO] Titanium.Android.Menu: OK!
[INFO] Titanium.Android.MenuItem: OK!
[INFO] Titanium.Android.Notification: OK!
[INFO] wakeLockOptions: OK!
[INFO] Titanium.Android.NotificationChannel: OK!
[INFO] Titanium.Android.NotificationManager: OK!
[INFO] Titanium.Android.PendingIntent: OK!
[INFO] Titanium.Android.QuickSettingsService: OK!
[INFO] showParams: OK!
[INFO] Titanium.Android.R: OK!
[INFO] Titanium.Android.RemoteViews: OK!
[INFO] Titanium.Android.Service: OK!
[INFO] Titanium.App.Android: OK!
[INFO] Titanium.App.Android.R: OK!
[INFO] Titanium.App: OK!
[INFO] launchOptions: OK!
[INFO] Titanium.App.Properties: OK!
[INFO] Titanium.App.iOS.BackgroundService: OK!
[INFO] Titanium.App.iOS.LocalNotification: OK!
[INFO] Titanium.App.iOS.SearchQuery: OK!
[INFO] Titanium.App.iOS.SearchableIndex: OK!
[INFO] Titanium.App.iOS.SearchableItem: OK!
[INFO] Titanium.App.iOS.SearchableItemAttributeSet: OK!
[INFO] Titanium.App.iOS.UserActivity: OK!
[INFO] Titanium.App.iOS.UserDefaults: OK!
[INFO] Titanium.App.iOS.UserNotificationAction: OK!
[INFO] Titanium.App.iOS.UserNotificationCategory: OK!
[INFO] Titanium.App.iOS.UserNotificationCenter: OK!
[INFO] UserNotificationCallbackResponse: OK!
[INFO] UserNotificationDictionary: OK!
[INFO] GetUserNotificationSettings: OK!
[INFO] Titanium.App.iOS: OK!
[INFO] NotificationParams: OK!
[INFO] UserNotificationAttachment: OK!
[INFO] LaunchOptionsType: OK!
[INFO] UserNotificationSettings: OK!
[INFO] Titanium.Blob: OK!
[INFO] ImageAsCroppedDict: OK!
[INFO] Titanium.BlobStream: OK!
[INFO] Titanium.Buffer: OK!
[INFO] Titanium.BufferStream: OK!
[INFO] Titanium.Calendar.Alert: OK!
[INFO] Titanium.Calendar.Attendee: OK!
[INFO] Titanium.Calendar: OK!
[INFO] EventsAuthorizationResponse: OK!
[INFO] Titanium.Calendar.Calendar: OK!
[INFO] Titanium.Calendar.Event: OK!
[INFO] Titanium.Calendar.RecurrenceRule: OK!
[INFO] daysOfTheWeekDictionary: OK!
[INFO] recurrenceEndDictionary: OK!
[INFO] Titanium.Calendar.Reminder: OK!
[INFO] Titanium.Codec: OK!
[INFO] EncodeNumberDict: OK!
[INFO] DecodeNumberDict: OK!
[INFO] EncodeStringDict: OK!
[INFO] DecodeStringDict: OK!
[INFO] Titanium.Contacts: OK!
[INFO] showContactsParams: OK!
[INFO] ContactsAuthorizationResponse: OK!
[INFO] Titanium.Contacts.Group: OK!
[INFO] Titanium.Contacts.Person: OK!
[INFO] Titanium.Database.DB: OK!
[INFO] BatchQueryError: OK!
[INFO] PossibleBatchQueryError: OK!
[INFO] Titanium.Database: OK!
[INFO] Titanium.Database.ResultSet: OK!
[INFO] ErrorResponse: OK!
[INFO] SuccessResponse: OK!
[INFO] FailureResponse: OK!
[INFO] Titanium.Event: OK!
[INFO] Titanium.Filesystem.File: OK!
[INFO] Titanium.Filesystem.FileStream: OK!
[INFO] Titanium.Filesystem: OK!
[INFO] RequestStorageAccessResult: OK!
[INFO] Titanium.Geolocation.Android: OK!
[INFO] Titanium.Geolocation.Android.LocationProvider: OK!
[INFO] Titanium.Geolocation.Android.LocationRule: OK!
[INFO] Titanium.Geolocation: OK!
[INFO] LocationResults: OK!
[INFO] LocationCoordinates: OK!
[INFO] LocationCoordinatesFloor: OK!
[INFO] ForwardGeocodeResponse: OK!
[INFO] HeadingResponse: OK!
[INFO] HeadingData: OK!
[INFO] LocationProviderDict: OK!
[INFO] ReverseGeocodeResponse: OK!
[INFO] GeocodedAddress: OK!
[INFO] LocationAuthorizationResponse: OK!
[INFO] Titanium.Gesture: OK!
[INFO] Titanium.IOStream: OK!
[INFO] Titanium.Locale: OK!
[INFO] Titanium.Media.Android: OK!
[INFO] MediaScannerResponse: OK!
[INFO] Titanium.Media.AudioPlayer: OK!
[INFO] TiMetadataItemType: OK!
[INFO] Titanium.Media.AudioRecorder: OK!
[INFO] Titanium.Media.Item: OK!
[INFO] Titanium.Media: OK!
[INFO] MusicLibraryOptionsType: OK!
[INFO] MusicLibraryResponseType: OK!
[INFO] MediaQueryType: OK!
[INFO] MediaQueryInfoType: OK!
[INFO] CameraOptionsType: OK!
[INFO] PhotoGalleryOptionsType: OK!
[INFO] CameraMediaItemType: OK!
[INFO] CropRectType: OK!
[INFO] PreviewRectType: OK!
[INFO] PreviewImageOptions: OK!
[INFO] PreviewImageError: OK!
[INFO] ScreenshotResult: OK!
[INFO] MediaAuthorizationResponse: OK!
[INFO] RequestCameraAccessResult: OK!
[INFO] RequestMusicLibraryAccessResult: OK!
[INFO] RequestPhotoGalleryAccessResult: OK!
[INFO] RouteDescription: OK!
[INFO] Titanium.Media.MusicPlayer: OK!
[INFO] Titanium.Media.Sound: OK!
[INFO] Titanium.Media.SystemAlert: OK!
[INFO] Titanium.Media.VideoPlayer: OK!
[INFO] ThumbnailResponse: OK!
[INFO] MovieSize: OK!
[INFO] Titanium.Module: OK!
[INFO] Titanium.Network.BonjourBrowser: OK!
[INFO] Titanium.Network.BonjourService: OK!
[INFO] Titanium.Network.Cookie: OK!
[INFO] Titanium.Network.HTTPClient: OK!
[INFO] SecurityManagerProtocol: OK!
[INFO] APSConnectionDelegate: OK!
[INFO] ReadyStatePayload: OK!
[INFO] Titanium.Network: OK!
[INFO] PushNotificationConfig: OK!
[INFO] PushNotificationSuccessArg: OK!
[INFO] PushNotificationErrorArg: OK!
[INFO] PushNotificationData: OK!
[INFO] Titanium.Network.Socket: OK!
[INFO] Titanium.Network.Socket.TCP: OK!
[INFO] ConnectedCallbackArgs: OK!
[INFO] ErrorCallbackArgs: OK!
[INFO] AcceptedCallbackArgs: OK!
[INFO] AcceptDict: OK!
[INFO] Titanium.Network.TCPSocket: OK!
[INFO] Titanium.Platform.Android: OK!
[INFO] Titanium.Platform.DisplayCaps: OK!
[INFO] Titanium.Platform: OK!
[INFO] CPU: OK!
[INFO] CPUTimes: OK!
[INFO] Titanium.Proxy: OK!
[INFO] Titanium.Stream: OK!
[INFO] CreateStreamArgs: OK!
[INFO] ReadCallbackArgs: OK!
[INFO] WriteCallbackArgs: OK!
[INFO] WriteStreamCallbackArgs: OK!
[INFO] PumpCallbackArgs: OK!
[INFO] Titanium: OK!
[INFO] CreateBufferArgs: OK!
[INFO] Titanium.UI.2DMatrix: OK!
[INFO] MatrixCreationDict: OK!
[INFO] Titanium.UI.3DMatrix: OK!
[INFO] Titanium.UI.ActivityIndicator: OK!
[INFO] Titanium.UI.ActivityIndicatorStyle: OK!
[INFO] Titanium.UI.AlertDialog: OK!
[INFO] Titanium.UI.Android: OK!
[INFO] Titanium.UI.Android.CardView: OK!
[INFO] Titanium.UI.Android.DrawerLayout: OK!
[INFO] Titanium.UI.Android.ProgressIndicator: OK!
[INFO] Titanium.UI.Android.SearchView: OK!
[INFO] Titanium.UI.Animation: OK!
[INFO] Attribute: OK!
[INFO] ParagraphAttribute: OK!
[INFO] Titanium.UI.AttributedString: OK!
[INFO] Titanium.UI.Button: OK!
[INFO] Titanium.UI.ButtonBar: OK!
[INFO] Titanium.UI.Clipboard: OK!
[INFO] ClipboardItemsType: OK!
[INFO] Titanium.UI.CoverFlowView: OK!
[INFO] Titanium.UI.DashboardItem: OK!
[INFO] Titanium.UI.DashboardView: OK!
[INFO] Titanium.UI.EmailDialog: OK!
[INFO] Font: OK!
[ERROR] ./apidoc/Titanium/UI/ImageView.yml
Titanium.UI.ImageView: found 1 error(s)!
	events
		error
			properties
				image
					Invalid key(s) in Titanium.UI.ImageView: optional

[INFO] Titanium.UI.Label: OK!
[INFO] Titanium.UI.ListItem: OK!
[INFO] ItemTemplate: OK!
[INFO] ViewTemplate: OK!
[INFO] ListDataItem: OK!
[INFO] RowActionType: OK!
[INFO] Titanium.UI.ListSection: OK!
[INFO] Titanium.UI.ListView: OK!
[INFO] ListViewAnimationProperties: OK!
[INFO] ListViewIndexEntry: OK!
[INFO] ListViewContentInsetOption: OK!
[INFO] ListViewMarkerProps: OK!
[INFO] ListViewEdgeInsets: OK!
[INFO] ListItemEventType: OK!
[INFO] SelectedItem: OK!
[INFO] Titanium.UI.MaskedImage: OK!
[INFO] Titanium.UI.Matrix2D: OK!
[INFO] Matrix2DCreationDict: OK!
[INFO] Titanium.UI.Matrix3D: OK!
[INFO] Matrix3DCreationDict: OK!
[INFO] Titanium.UI.NavigationWindow: OK!
[INFO] Titanium.UI.Notification: OK!
[INFO] Titanium.UI.OptionDialog: OK!
[INFO] hideParams: OK!
[INFO] Titanium.UI.Picker: OK!
[INFO] Titanium.UI.PickerColumn: OK!
[INFO] Titanium.UI.PickerRow: OK!
[INFO] Titanium.UI.ProgressBar: OK!
[INFO] Titanium.UI.RefreshControl: OK!
[INFO] Titanium.UI.ScrollView: OK!
[INFO] contentOffsetOption: OK!
[INFO] zoomScaleOption: OK!
[INFO] ScrollViewAnimationProperties: OK!
[INFO] Titanium.UI.ScrollableView: OK!
[INFO] Titanium.UI.SearchBar: OK!
[INFO] Titanium.UI.ShortcutItem: OK!
[INFO] Titanium.UI.Slider: OK!
[INFO] Titanium.UI.Switch: OK!
[INFO] Titanium.UI.Tab: OK!
[INFO] TabIconInsets: OK!
[INFO] Titanium.UI.TabGroup: OK!
[INFO] Titanium.UI.TabbedBar: OK!
[INFO] Titanium.UI.TableView: OK!
[INFO] TableViewAnimationProperties: OK!
[INFO] TableViewIndexEntry: OK!
[INFO] TableViewContentInsetOption: OK!
[INFO] TableViewEdgeInsets: OK!
[INFO] SelectedRowObject: OK!
[INFO] Titanium.UI.TableViewRow: OK!
[INFO] Titanium.UI.TableViewSection: OK!
[INFO] Titanium.UI.TextArea: OK!
[INFO] textAreaSelectedParams: OK!
[INFO] Titanium.UI.TextField: OK!
[INFO] textFieldSelectedParams: OK!
[INFO] TextFieldPadding: OK!
[INFO] Titanium.UI.Toolbar: OK!
[INFO] Titanium.UI: OK!
[INFO] Titanium.UI.View: OK!
[INFO] Point: OK!
[INFO] Gradient: OK!
[INFO] GradientColorRef: OK!
[INFO] Dimension: OK!
[INFO] AnimationOption: OK!
[INFO] ViewPadding: OK!
[INFO] Titanium.UI.WebView: OK!
[INFO] SnapshotResult: OK!
[INFO] BackForwardList: OK!
[INFO] BackForwardListItem: OK!
[INFO] OnLinkURLResponse: OK!
[INFO] Titanium.UI.Window: OK!
[INFO] openWindowParams: OK!
[INFO] windowToolbarParam: OK!
[INFO] closeWindowParams: OK!
[INFO] titleAttributesParams: OK!
[INFO] shadowDict: OK!
[INFO] Titanium.UI.iOS.AdView: OK!
[INFO] Titanium.UI.iOS.AlertDialogStyle: OK!
[INFO] Titanium.UI.iOS.AnchorAttachmentBehavior: OK!
[INFO] Titanium.UI.iOS.AnimationStyle: OK!
[INFO] Titanium.UI.iOS.Animator: OK!
[INFO] Titanium.UI.iOS.ApplicationShortcuts: OK!
[INFO] ShortcutParams: OK!
[INFO] Titanium.UI.iOS.Attribute: OK!
[INFO] Titanium.UI.iOS.AttributedString: OK!
[INFO] Titanium.UI.iOS.BlurView: OK!
[INFO] Titanium.UI.iOS.CollisionBehavior: OK!
[INFO] BoundaryIdentifier: OK!
[INFO] ReferenceInsets: OK!
[INFO] Titanium.UI.iOS.CoverFlowView: OK!
[INFO] CoverFlowImageType: OK!
[INFO] Titanium.UI.iOS.DocumentViewer: OK!
[INFO] DocumentViewerOptions: OK!
[INFO] Titanium.UI.iOS.DynamicItemBehavior: OK!
[INFO] Titanium.UI.iOS.FeedbackGenerator: OK!
[INFO] Titanium.UI.iOS.GravityBehavior: OK!
[INFO] Titanium.UI.iOS.ListViewCellSelectionStyle: OK!
[INFO] Titanium.UI.iOS.ListViewScrollPosition: OK!
[INFO] Titanium.UI.iOS.ListViewStyle: OK!
[INFO] Titanium.UI.iOS.LivePhoto: OK!
[INFO] Titanium.UI.iOS.LivePhotoView: OK!
[INFO] Titanium.UI.iOS.MenuPopup: OK!
[INFO] MenuPopupShowParams: OK!
[INFO] MenuPopupHideParams: OK!
[INFO] Titanium.UI.iOS.NavigationWindow: OK!
[INFO] Titanium.UI.iOS.PreviewAction: OK!
[INFO] Titanium.UI.iOS.PreviewActionGroup: OK!
[INFO] Titanium.UI.iOS.PreviewContext: OK!
[INFO] Titanium.UI.iOS.ProgressBarStyle: OK!
[INFO] Titanium.UI.iOS.PushBehavior: OK!
[INFO] Titanium.UI.iOS.RowAnimationStyle: OK!
[INFO] Titanium.UI.iOS.ScrollIndicatorStyle: OK!
[INFO] Titanium.UI.iOS.SnapBehavior: OK!
[INFO] Titanium.UI.iOS.SplitWindow: OK!
[INFO] animationOption: OK!
[INFO] Titanium.UI.iOS.StatusBar: OK!
[INFO] Titanium.UI.iOS.Stepper: OK!
[INFO] Titanium.UI.iOS.SystemButton: OK!
[INFO] Titanium.UI.iOS.SystemButtonStyle: OK!
[INFO] Titanium.UI.iOS.SystemIcon: OK!
[INFO] Titanium.UI.iOS.TabbedBar: OK!
[INFO] BarItemType: OK!
[INFO] Titanium.UI.iOS.TableViewCellSelectionStyle: OK!
[INFO] Titanium.UI.iOS.TableViewScrollPosition: OK!
[INFO] Titanium.UI.iOS.TableViewStyle: OK!
[INFO] Titanium.UI.iOS.Toolbar: OK!
[INFO] Titanium.UI.iOS.ViewAttachmentBehavior: OK!
[INFO] Titanium.UI.iOS.WebViewConfiguration: OK!
[INFO] WebViewPreferencesObject: OK!
[INFO] Titanium.UI.iOS.WebViewDecisionHandler: OK!
[INFO] Titanium.UI.iOS.WebViewProcessPool: OK!
[INFO] Titanium.UI.iOS: OK!
[INFO] transitionAnimationParam: OK!
[INFO] Titanium.UI.iPad.DocumentViewer: OK!
[INFO] Titanium.UI.iPad.Popover: OK!
[INFO] PopoverParams: OK!
[INFO] Titanium.UI.iPad.SplitWindow: OK!
[INFO] Titanium.UI.iPad: OK!
[INFO] Titanium.UI.iPhone.ActivityIndicatorStyle: OK!
[INFO] Titanium.UI.iPhone.AlertDialogStyle: OK!
[INFO] Titanium.UI.iPhone.AnimationStyle: OK!
[INFO] Titanium.UI.iPhone.ListViewCellSelectionStyle: OK!
[INFO] Titanium.UI.iPhone.ListViewScrollPosition: OK!
[INFO] Titanium.UI.iPhone.ListViewSeparatorStyle: OK!
[INFO] Titanium.UI.iPhone.ListViewStyle: OK!
[INFO] Titanium.UI.iPhone.ProgressBarStyle: OK!
[INFO] Titanium.UI.iPhone.RowAnimationStyle: OK!
[INFO] Titanium.UI.iPhone.ScrollIndicatorStyle: OK!
[INFO] Titanium.UI.iPhone.SystemButton: OK!
[INFO] Titanium.UI.iPhone.SystemButtonStyle: OK!
[INFO] Titanium.UI.iPhone.SystemIcon: OK!
[INFO] Titanium.UI.iPhone.TableViewCellSelectionStyle: OK!
[INFO] Titanium.UI.iPhone.TableViewScrollPosition: OK!
[INFO] Titanium.UI.iPhone.TableViewSeparatorStyle: OK!
[INFO] Titanium.UI.iPhone.TableViewStyle: OK!
[INFO] Titanium.UI.iPhone: OK!
[INFO] Titanium.Utils: OK!
[INFO] Titanium.WatchSession: OK!
[INFO] MessageReply: OK!
[INFO] Titanium.XML.Attr: OK!
[INFO] Titanium.XML.CDATASection: OK!
[INFO] Titanium.XML.CharacterData: OK!
[INFO] Titanium.XML.Comment: OK!
[INFO] Titanium.XML.DOMImplementation: OK!
[INFO] Titanium.XML.Document: OK!
[INFO] Titanium.XML.DocumentFragment: OK!
[INFO] Titanium.XML.DocumentType: OK!
[INFO] Titanium.XML.Element: OK!
[INFO] Titanium.XML.Entity: OK!
[INFO] Titanium.XML.EntityReference: OK!
[INFO] Titanium.XML.NamedNodeMap: OK!
[INFO] Titanium.XML.Node: OK!
[INFO] Titanium.XML.NodeList: OK!
[INFO] Titanium.XML.Notation: OK!
[INFO] Titanium.XML.ProcessingInstruction: OK!
[INFO] Titanium.XML.Text: OK!
[INFO] Titanium.XML: OK!
[INFO] Titanium.Yahoo: OK!
[INFO] YQLResponse: OK!
[INFO] Dictionary: OK!
[INFO] Titanium.Namespace.Name: OK!
[ERROR] Found 1 error(s)!
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! titanium-mobile@9.0.0 lint:docs: `tdoc-validate ./apidoc`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the titanium-mobile@9.0.0 lint:docs script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/build/.npm/_logs/2019-11-20T10_52_33_826Z-debug.log
ERROR: "lint:docs" exited with 1.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! titanium-mobile@9.0.0 lint: `npm-run-all --parallel lint:**`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the titanium-mobile@9.0.0 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/build/.npm/_logs/2019-11-20T10_52_34_082Z-debug.log
npm ERR! Test failed.  See above for more details.

```
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d5426b36033ac52c8b1ccdcde26708a1d5f93359
</p>

--- a/apidoc/Titanium/App/App.yml
+++ b/apidoc/Titanium/App/App.yml
@@ -256,6 +256,7 @@ events:
     properties:
       - name: state
         summary: Proximity state value.
+        type: Boolean
     platforms: [iphone, android]
     since: {android: "3.3.0"}
 
@@ -267,31 +268,41 @@ events:
     properties:
       - name: message
         summary: The error message.
+        type: String
       - name: line
         summary: The line where the error occurred.
+        type: Number
       - name: sourceId
         summary: A unique identification for the source file.
+        type: Number
         platforms: [iphone, ipad]
       - name: type
         summary: The type of error.
+        type: String
         platforms: [iphone, ipad]
       - name: sourceURL
         summary: The URL to the source file.
+        type: String
         platforms: [iphone, ipad]
       - name: backtrace
         summary: The backtrace of function calls when the error occurred.
+        type: String
         platforms: [iphone, ipad]
       - name: title
         summary: The title for the error.
+        type: String
         platforms: [android]
       - name: sourceName
         summary: The name of the source file.
+        type: String
         platforms: [android]
       - name: lineSource
         summary: The line source reference.
+        type: String
         platforms: [android]
       - name: lineOffset
         summary: The offset on the line where the error occurred.
+        type: Number
         platforms: [android]
     platforms: [android, iphone, ipad]
     since: '4.1.0'

--- a/apidoc/Titanium/Geolocation/Geolocation.yml
+++ b/apidoc/Titanium/Geolocation/Geolocation.yml
@@ -307,6 +307,7 @@ events:
     properties:
       - name: code
         summary: If `success` is `false`, the error code is available.
+        type: Number
         platforms: [android, iphone, ipad]
 
       - name: success

--- a/apidoc/Titanium/Media/VideoPlayer.yml
+++ b/apidoc/Titanium/Media/VideoPlayer.yml
@@ -388,6 +388,7 @@ events:
     properties:
       - name: sourceType
         summary: New `sourceType` value.
+        type: Number
         constants: Titanium.Media.VIDEO_SOURCE_TYPE_*
     since: '3.0.0'
 

--- a/apidoc/Titanium/Network/BonjourBrowser.yml
+++ b/apidoc/Titanium/Network/BonjourBrowser.yml
@@ -29,7 +29,7 @@ events:
     properties:
       - name: services
         summary: An array of BonjourService objects corresponding to currently available services.  If you cache this value, including using it as table data, be aware that it could become out of date at any time due to the asynchronous nature of Bonjour service discovery.
-        type: Array
+        type: Array<Titanium.Network.BonjourService>
     since: '3.0.0'
 
 properties:

--- a/apidoc/Titanium/UI/ImageView.yml
+++ b/apidoc/Titanium/UI/ImageView.yml
@@ -123,7 +123,6 @@ events:
       - name: image
         summary: URL of the image that failed to load.
         type: String
-        optional: true
 
   - name: pause
     summary: Fired when the animation pauses.

